thunder/tests/distributed/test_fsdp.py

```diff
@@ -917,5 +976,32 @@ def test_fsdp_transformer_engine(executor, devices, dtype, thunder_fsdp_strategy
     pass
 
 
+@instantiate(
+    dtypes=(thunder.float32,),
+    num_devices=2,
+    devicetypes=(devices.DeviceType.CUDA,),
+    executors=(TorchExecutor,),
+    decorators=(
+        # NOTE: ddp_wrapper
+        pytest.mark.parametrize(
+            "thunder_fsdp_strategy_and_bucketing",
+            (
+                (FSDPType.ZERO3, FSDPBucketingStrategy.LAYER),
+                (FSDPType.ZERO3, FSDPBucketingStrategy.BLOCK),
+            ),
+        ),
+        pytest.mark.skipif(not TE_AVAILABLE, reason="TransformerEngine is not installed."),
+        pytest.mark.skipif(not is_fp8_supported, reason=fp8_support_reason),
+        # See NOTE: Setting `NVTE_TORCH_COMPILE`
+        # NOTE: We don't pass `clear=True` to `unittest.mock.patch.dict` as that may clear paths
+        # from environment leading to picking up of incorrect dependencies in the spawned process.
+        unittest.mock.patch.dict(os.environ, {"NVTE_TORCH_COMPILE": "0"}),
+    ),
+)
+@distributed_wrapper("test_fsdp_transformer_engine_bucketing", _test_fsdp_transformer_engine_bucketing)
+def test_fsdp_transformer_engine_bucketing(executor, devices, dtype, thunder_fsdp_strategy_and_bucketing):
```

Comment:
This test fails on the current main branch, right?

Reply:
Yes, that is correct. It fails with

```python
        # Raises the first assertion if any occurred
        root_results = results[0]
        if len(root_results) > 0:
>           raise (root_results[0])
E           KeyError: 'ctx_te_1'
```